compiler/src/iree/compiler/Dialect/VMVX/Transforms/ResolveBufferDescriptors.cpp

```diff
@@ -184,66 +303,95 @@ struct FromHalInterfaceBindingSubspan
-  LogicalResult matchAndRewrite(GetBufferDescriptorOp op,
+struct ResolveExtractMetadataFromHalInterfaceBindingSubspan
+    : public OpRewritePattern<memref::ExtractStridedMetadataOp> {
+  using OpRewritePattern<memref::ExtractStridedMetadataOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(memref::ExtractStridedMetadataOp op,
                                 PatternRewriter &rewriter) const override {
-    auto alloca = op.getSource().getDefiningOp<memref::AllocaOp>();
-    if (!alloca) return failure();
-    auto memRefType = alloca.getResult().getType().cast<MemRefType>();
-    if (!memRefType.getLayout().isIdentity()) {
-      return rewriter.notifyMatchFailure(op, "not identity allocation");
-    }
+    auto binding =
+        op.getSource()
+            .template getDefiningOp<IREE::HAL::InterfaceBindingSubspanOp>();
+    if (!binding) return failure();
+    auto memRefType = binding.getResult().getType().template cast<MemRefType>();
+    if (memRefType.getRank() < 1) return failure();
 
     auto loc = op.getLoc();
-    IndexSet indexSet(loc, rewriter);
-
-    // Replace the op with values:
-    //   base_buffer: The subspan result
-    //   offset: byte offset from subspan divided by element type size
-    //   sizes: static and dynamic sizes from the subspan
-    //   strides: identity strides
     int rank = memRefType.getRank();
+    DescriptorInfo resultDescriptor;
 
     // Compute sizes.
-    SmallVector<Value> sizes;
-    auto dynamicDimIt = alloca.getDynamicSizes().begin();
+    auto dynamicDimIt = binding.getDynamicDims().begin();
     for (int i = 0; i < rank; ++i) {
       if (memRefType.isDynamicDim(i)) {
-        sizes.push_back(*dynamicDimIt);
+        resultDescriptor.sizes.push_back(*dynamicDimIt);
         dynamicDimIt++;
       } else {
-        sizes.push_back(rewriter.create<arith::ConstantIndexOp>(
-            loc, memRefType.getDimSize(i)));
+        resultDescriptor.sizes.push_back(
+            rewriter.getIndexAttr(memRefType.getDimSize(i)));
       }
+    }
 
-      // Replace as we go.
-      op.getSizes()[i].replaceAllUsesWith(sizes.back());
+    // Strides.
+    resultDescriptor.strides =
+        getStridesFromSizes(rewriter, loc, resultDescriptor.sizes);
+    resultDescriptor.offset = rewriter.getIndexAttr(0);
+
+    replaceOffsetSizesAndStridesWith(rewriter, op, resultDescriptor);
+
+    // Base buffer. Use a 1D memref for hal.interface.binding.subspan.
+    AffineMap mulMap = getMulMap(rewriter.getContext());
+    OpFoldResult linearizedMemrefSize = rewriter.getIndexAttr(1);
+    for (auto size : resultDescriptor.sizes) {
+      linearizedMemrefSize = makeComposedFoldedAffineApply(
+          rewriter, loc, mulMap, {linearizedMemrefSize, size});
     }
+    SmallVector<int64_t> staticLinearShape;
+    SmallVector<Value> dynamicLinearShape;
+    dispatchIndexOpFoldResult(linearizedMemrefSize, dynamicLinearShape,
+                              staticLinearShape);
+    Value linearInterfaceBinding =
```

Comment:
rewriter should set the insertion point to `binding` here

Reply:
You are right. Let me fix those.